File: 프로그래머스/연습문제/12915_문자열_내_마음대로_정렬하기.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/12915

// Todo: sort 커스텀

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    sort(strings.begin(), strings.end(), [n](const string& a, const string& b) {
        if (a[n] != b[n]) {
            return a[n] < b[n];
        }

        return a < b;
    });

    return strings;
}

File: 프로그래머스/연습문제/12915_문자열_내_마음대로_정렬하기.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12915

// Todo: Arrays.asList(), Collections.sort()

import java.util.*;

class Solution {
    public List<String> solution(String[] strings, int n) {
        List<String> stringList = Arrays.asList(strings);
        Collections.sort(stringList, (a, b) -> {
            if (a.charAt(n) != b.charAt(n)) {
                return Character.compare(a.charAt(n), b.charAt(n));
            }
            return a.compareTo(b);
        });
        return stringList;
    }
}